timestamp.go

```diff
@@ -22,28 +22,37 @@ type Timestamp time.Time
 // MarshalJSON defines how encoding/json marshals the object to JSON,
 // the result is a string of the UNIX timestamp
 func (t Timestamp) MarshalJSON() ([]byte, error) {
-	ts := t.Time().Unix()
+	var ts int64
+	if t.Time().Unix() < 1e12 {
```

Comment:
I don't think this will work.
.Unix() will always give you seconds, so the condition will always be True - so we're back to serializing everything to seconds.
See here:
https://go.dev/play/p/DY6vkNIBAmP?v=goprev
Anyway, I don't think the IF is necessary just remove it and do ts = t.Time().UnixMilli() as before